core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -2773,6 +2773,27 @@ protected RecordCursorFactory generateFactory(QueryModel selectQueryModel, SqlEx
         return codeGenerator.generate(selectQueryModel, executionContext);
     }
 
+    RecordCursorFactory generateWithRetries(
+            @Transient QueryModel initialQueryModel,
+            @Transient SqlExecutionContext executionContext
+    ) throws SqlException {
+        QueryModel queryModel = initialQueryModel;
+        int remainingRetries = TableReferenceOutOfDateException.MAX_RETRY_ATTEMPTS;
+        for (; ; ) {
+            try {
+                return generateFactory(queryModel, executionContext);
+            } catch (TableReferenceOutOfDateException e) {
+                if (--remainingRetries < 0) {
+                    throw SqlException.$(0, "unable to generate query plan, table structure changed too many times, [q=`").put(queryModel).put('`').put(", fd=").put(executionContext.getRequestFd()).put(']');
```

Comment:
In other places we simply use `e.getFlyweightMessage()`. Why do we need a different exception message here?

Reply:
good question. I don't think there is a good reason.